src/antares/service/local_services/binding_constraint_local.py

```diff
@@ -103,20 +95,76 @@ def _check_if_empty_ts(time_step: BindingConstraintFrequency, time_series: Optio
         time_series_length = (365 * 24 + 24) if time_step == BindingConstraintFrequency.HOURLY else 366
         return time_series if time_series is not None else pd.DataFrame(np.zeros([time_series_length, 1]))
 
-    def _write_binding_constraint_ini(self) -> None:
-        binding_constraints_ini_content = {
-            idx: idx_constraint.local_properties.list_ini_fields
-            for idx, idx_constraint in enumerate(self.binding_constraints.values())
-        }
-        self.ini_file.ini_dict = binding_constraints_ini_content
+    def _write_binding_constraint_ini(
+        self,
+        properties: BindingConstraintProperties,
+        constraint_name: str,
+        constraint_id: str,
+        terms: Optional[list[ConstraintTerm]] = None,
+    ) -> None:
+        """
+        Write a single binding constraint to the INI file, reconstructing a full BindingConstraintPropertiesLocal instance.
+
+        Args:
+            properties (BindingConstraintProperties): Basic properties of the binding constraint.
+            constraint_name (str): The name of the constraint.
+            constraint_id (str): The ID of the constraint.
+            terms (dict[str, ConstraintTerm], optional): Terms applying to the binding constraint. Defaults to None.
+        Raises:
+        BindingConstraintCreationError: If a binding constraint with the same name already exists in the INI file.
+        """
+
+        current_ini_content = self.ini_file.ini_dict or {}
+
+        if constraint_name in current_ini_content:
+            if terms in [None, {}]:
```

Comment:
how can we have terms = {} as you typed it as a list ?
Also Pycharm tells me to merge the 2 if statements 
And also I don't get why you check the terms to know if you raise or not ? It seems it's because it one case you're updating and in the other creating. If so, it should be more explicit or separated in 2 methods